### ISSUE TITLE
Unable to rerun example code

### ISSUE THREAD

**Issue by davecan98** [resttime/cl-liballegro#39]
Hi,

I'm new to Common Lisp in general, so apologies if I'm asking an obvious question.

I'm running the example code below which displays an example window. If I run the main function from the repl, I can see the window display fine. 

```
(defclass window (al:system)
  ((previous-key :initform "Nothing" :accessor previous-key))
  (:default-initargs :title ""
    :width 800 :height 600
    :logic-fps 1
    :display-flags '(:windowed :opengl :resizable)
    :display-options '((:sample-buffers 1 :suggest)
		       (:samples 4 :suggest))))

(defmethod al:update ((sys window))
  (print 'one-logic-frame))

(defmethod al:render ((sys window))
  (al:clear-to-color (al:map-rgb 20 150 100))
  (al:flip-display))

(defmethod al:key-down-handler ((sys window))
  (let ((keyboard (cffi:mem-ref (al:event sys) '(:struct al:keyboard-event))))
    (print (getf keyboard 'al::keycode))
    (setf (previous-key sys) (getf keyboard 'al::keycode))))

(defun main ()
  (al:run-system (make-instance 'window)))

```

However, when I close down the window and try to run main again, all I see is the following being evaluated, and nothing else happens. Is there some cleanup functions I'm meant to use before I call main again?

```
#<SIMPLE-TASKS:CALL-TASK :FUNC #<FUNCTION (LAMBDA ()
                                            :IN
                                            CL-LIBALLEGRO:RUN-SYSTEM) {53FC83DB}> :STATUS :SCHEDULED {1002A50703}>

```

Thanks

**Comment by resttime** [comment]
No problem, you've found a bug which I've verified so thanks. 

Cause is in some code introduced for cross-platform compatibility on OSX dealing with threads. I think I've been able to fix it just gotta verify there aren't any other unintended behaviours. Fix will be released soon after I get back.

**Comment by resttime** [comment]
Should be fixed now with: 62e006d9c63f5cfb34c5f4b5bb66f327488b8584

If using quicklisp, cloning this repo into `quicklisp/local-projects` should allow using the latest changes. Otherwise it'll get picked up in the next quicklisp release.the interface is something that's technically optional, so rest of the library should be ok . Thanks for the bug report.

**Comment by resttime** [comment]
Also the rest of the library should be okay as the issue is just in the optional lispy inteface
